src/modules/languages/c.nix

```diff
@@ -2,20 +2,42 @@
 
 let
   cfg = config.languages.c;
+
+  # Default to GCC if no compiler is explicitly enabled
+  useGcc = cfg.gcc.enable || !cfg.clang.enable;
+  useClang = cfg.clang.enable;
 in
 {
   options.languages.c = {
     enable = lib.mkEnableOption "tools for C development";
+
+    clang.enable = lib.mkEnableOption "clang";
+    gcc.enable = lib.mkEnableOption "GCC";
   };
 
   config = lib.mkIf cfg.enable {
-    packages = with pkgs; [
-      stdenv
+    languages.c = {
+      clang.enable = lib.mkDefault false;
```

Comment:
Should be the default to use the platform-default for compilers i.e. gcc on Linux and clang on macOS? From my experience many project break if you try to use gcc on macOS because they assume clang specific features.